vaporetto/src/model.rs

```diff
@@ -8,27 +8,34 @@ use bincode::{Decode, Encode};
 
 use crate::dict_model::{DictModel, WordWeightRecord};
 use crate::errors::{Result, VaporettoError};
-use crate::ngram_model::NgramModel;
-use crate::tag_model::TagModel;
+use crate::ngram_model::{NgramModel, TagNgramModel};
 use crate::utils::VecWriter;
 
 /// Magic number.
-const MODEL_MAGIC: &[u8] = b"VaporettoTokenizer 0.4.0\n";
+const MODEL_MAGIC: &[u8] = b"VaporettoTokenizer 0.5.0\n";
 
-/// Model data.
-pub struct Model {
-    pub(crate) data: ModelData,
+#[derive(Debug, Decode, Encode)]
+pub struct TagModel {
+    pub(crate) token: String,
+    pub(crate) tags: Vec<Vec<String>>,
+    pub(crate) char_ngram_model: TagNgramModel<String>,
+    pub(crate) type_ngram_model: TagNgramModel<Vec<u8>>,
+    pub(crate) bias: Vec<i32>,
```

Comment:
It would be helpful to add comments to describe the dimensions of tag-weight arrays.

Reply:
d468842